export.py

```diff
@@ -1916,18 +1986,83 @@ def cache_motion(scene, rpass, objects=None):
                 scene.frame_set(origframe - 1, 1.0 + seg)
             else:
                 scene.frame_set(origframe, seg)
-
+            
             for name in instance_names:
                 get_transform(instances[name], seg)
 
             for name in data_names:
                 get_deformation(data_blocks[name], seg, scene)
 
     scene.frame_set(origframe, 0)
-
+    elapsed = time.perf_counter() - start
+    debug("warning", "...caching transformations and deformations took [%f]"%elapsed)
+    
+    try:
+
+        # set transforming property of instances correctly
+        debug("debug", "Checking for transformations")
+        start = time.perf_counter()
+        if(instance_names is not None and len(instance_names)>0):
+            for instance_name in instance_names:
+                instance = instances[instance_name]
+                last_matrix = None
+                instance.transforming = False
+                for subframe, matrix in instance.motion_data:
+                        if(last_matrix is None):
+                            last_matrix=matrix
+                        elif(last_matrix!=matrix):
+                            instance.transforming = True
+                            break
+                        last_matrix = matrix
+            
+            for instance_name in instance_names:
+                instance = instances[instance_name]
+                if(not instance.transforming):
+                    instance.transforming = is_transforming_recursive_brute_force(instance, instances)
+                if(not instance.transforming):
+                    instance.motion_data.clear()
+                    #debug("debug", "Object [%s] is not transforming." % instance.name)
+        # endif 
+        elapsed = time.perf_counter() - start
+        debug("warning", "...checking transformations took [%f]"%elapsed)
+        # cache whether a mesh is deforming or not
+        debug("warning", "Checking for deformation")
+        IsMeshDeforming.clear()
+        start = time.perf_counter()
+        if(data_names is not None):
+            for data_block_name in data_names:
+                data_block = data_blocks[data_block_name]
+                last_mesh = None
+                if(not data_block.do_export):
+                    debug("debug", "Deformations for mesh [%s] is not checked as file was not dirty and no need to make a new export."% data_block.name)                
+                elif(data_block.deforming and not is_deforming_fluid(data_block.data)):
```

Comment:
Just to let you know I'm revisiting your checkin!  I'm basically going to take your code and more it around a little